src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplCaseInvalidateTest.java

```diff
@@ -105,4 +113,17 @@ public void shouldRejectCaseNotFound() throws Exception {
     InvalidateCaseRequestDTO dto = CaseServiceFixture.createInvalidateCaseRequestDTO();
     target.invalidateCase(dto);
   }
+
+  @Test(expected = Exception.class)
+  public void shouldRejectCaseOfTypeCE() throws Exception {
+    InvalidateCaseRequestDTO dto = CaseServiceFixture.createInvalidateCaseRequestDTO();
```

Comment:
Since the message returned in the JIRA is so specific , we should test that . The best way of doing this I have found is to use the JUnit 4.13 assertThrows.
Here is an example (from CaseServiceImplGetCaseByCaseRefTest ):
```
   import static org.junit.Assert.assertThrows;

    CTPException e =
        assertThrows(CTPException.class, () -> target.getCaseByCaseReference(caseRef, dto));
    assertEquals("Invalid Case Reference", e.getMessage());
```
In your case the message tested would start with **"All CE address wil be validated ....**

Reply:
that's handy to know, thanks - done